simple-wp-helpdesk/src/Rest/Bootstrap.php

```
@@ -0,0 +1,205 @@
+		}
+
+		// REMOTE_ADDR is a server-provided IP; sanitize_text_field strips
+		// anything that wouldn't survive a hash_hmac call. Fallback to 0.0.0.0
+		// when the server didn't populate it (CLI, tests, exotic SAPIs).
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotValidated, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+		$raw_ip = isset( $_SERVER['REMOTE_ADDR'] ) && is_string( $_SERVER['REMOTE_ADDR'] )
+			// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotValidated, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+			? sanitize_text_field( (string) $_SERVER['REMOTE_ADDR'] )
+			: '0.0.0.0';
+		$hashed = hash_hmac( 'sha256', $raw_ip, wp_salt( 'nonce' ) );
+		$state  = RateLimiter::check( 'swh_rl_ip_' . $hashed, self::PUBLIC_IP_RATE_LIMIT );
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'SWH\Rest\RateLimiter' in method 'settings_public_permission'. (undefined)

(StaticAccess)